Examples/Cocoapods/SpeechToText/Speech/AudioController.swift

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2019, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import Foundation
+import AVFoundation
+
+protocol AudioControllerDelegate {
+  func processSampleData(_ data: Data) -> Void
+}
+
+class AudioController {
+  var remoteIOUnit: AudioComponentInstance? // optional to allow it to be an inout argument
+  var delegate: AudioControllerDelegate!
+  
+  static var sharedInstance = AudioController()
+  
+  deinit {
+    AudioComponentInstanceDispose(remoteIOUnit!)
+  }
+  
+  func prepare(specifiedSampleRate: Int) -> OSStatus {
```

Comment:
I know this is an example and the following isn't really the subject of the example but it would be useful to provide more of a description as to what each part of this is doing. Breaking it up into a couple of functions would also help readability!